### PR TITLE
Create Swift 3 runtime with KituraNet.  Update CLI, add CLI and container tests

### PR DESCRIPTION
**Pull request by paulcastro**
Supercedes PR #308 .

### CODE REVIEW

File: common/scala/src/whisk/core/entity/Exec.scala [hunk 1]
```diff
@@ -164,7 +175,14 @@ protected[core] object Exec
                     }
                     JavaExec(jar, main)
 
-                case _ => throw new DeserializationException(s"'kind' must be one of {${Exec.NODEJS},${Exec.BLACKBOX},${Exec.SWIFT},${Exec.JAVA}}")
+                case  Exec.SWIFT3 =>
+                    val code: String = obj.getFields("code") match {
+                        case Seq(JsString(c)) => c
+                        case _                => throw new DeserializationException(s"'code' must be a string defined in 'exec' for '${Exec.SWIFT}, ${Exec.SWIFT3}' actions")
```
Comment (psuter) [inline common/scala/src/whisk/core/entity/Exec.scala]: Why mention `swift` here? (as opposed to just `swift3`?)

File: tools/cli/wskaction.py [hunk 2]
```diff
@@ -170,7 +174,10 @@ def getExec(self, args, props):
             exe = self.getActionExec(args, props, pipeAction)
         elif args.artifact is not None and os.path.isfile(args.artifact):
             contents = open(args.artifact, 'rb').read()
-            if args.artifact.endswith('.swift'):
+            if args.kind in ['swift:3','swift:3.0','swift:3.0.0']:
```
Comment (psuter) [inline tools/cli/wskaction.py]: Why does the CLI expect the `runtime:version` syntax but this isn't reflected in the JSON object? I understand that later in code we can't have a class called e.g. `Swift:3Action`, but I would think at least the format for `--kind` and `kind` should match.

File: mobile/Linux/kituraTest.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import KituraNet
```
Comment (psuter) [inline mobile/Linux/kituraTest.swift]: Is this file referred to anywhere?

File: common/scala/src/whisk/core/entity/Exec.scala (hunk 1 above)
Comment (paulcastro) [inline common/scala/src/whisk/core/entity/Exec.scala]: Changed.

File: mobile/Linux/kituraTest.swift (hunk 3 above)
Comment (paulcastro) [inline mobile/Linux/kituraTest.swift]: Removed

File: tools/cli/wskaction.py (hunk 2 above)
Comment (paulcastro) [inline tools/cli/wskaction.py]: I'm fine changing the syntax to something more consistent. @sjfink , what do you think?

Comment (sjfink) [inline tools/cli/wskaction.py]: I'm not sure what syntax change is being proposed.

Prior to this PR, we supported "kind" = "nodejs", "docker", "java", "swift"

This PR was going to add at least "swift:3" to that list.

I don't want this PR to be a breaking API change, so I don't want to force kind to always be "name:version" , at least for now.

Clear?   Did I misunderstand the question?

Comment (psuter) [inline tools/cli/wskaction.py]: The question is basically about consistency between the REST API (where currently creating a Swift3 action means having a field `"kind": "swift3"` in the POST payload) and the CLI, where currently creating a Swift3 action means using `--kind=swift:3`. They're inconsistent.

Comment (sjfink) [inline tools/cli/wskaction.py]: Oh - ok.  then I agree, both the API and CLI should use "kind" : "swift:3"

Comment (paulcastro) [inline tools/cli/wskaction.py]: Change made

**Comment by paulcastro** [comment]
irisPlayground build id is 273.

**Pull request closed without merging**